kubernetes/Scaling_CoreDNS.md

```diff
@@ -0,0 +1,99 @@
+# Scaling CoreDNS in Kubernetes Clusters
+
+
+The following is a guide for tuning CoreDNS in Kubernetes clusters.
+
+
+## Memory and Pods
+
+In large scale Kubernetes clusters, CoreDNS's memory usage is predominantly affected by the number of Pods and Services in the cluster. Other factors include the size of the filled DNS answer cache, and the rate of queries received (QPS) per CoreDNS instance.
+
+### With default CoreDNS settings
+
+To estimate the amount of memory required for a CoreDNS instance (using default settings), you can use the following formula:
+
+>  MB required (default settings) = (Pods + Services) * 0.000858 + 49.1
+
```

Comment:
I would simplify to 
>  MB required (default settings) = (Pods + Services) * 0.001 + 50

so it is easy to compute.

Reply:
well maybe people can adjust themselves if they wish...

Reply:
@fturib I simplified the math slightly expressing as division instead of multiply by a number < 0.
I also noticed the offsets didn't match the graph... they were old offsets before I included the 5 mb operating buffer.

Reply:
Why can't we point to metrics? We should have this and if not we should make an effort to make this available.